plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.test.ts

```diff
@@ -115,8 +115,34 @@ describe('DatabaseTaskStore', () => {
       eventBody: { message },
     });
 
-    const { tasks } = await store.list({ status: 'open' });
+    const { tasks, totalTasks } = await store.list({
+      filters: { status: ['open'] },
+    });
+    expect(tasks.length).toBe(1);
+    expect(totalTasks).toBe(1);
+    expect(tasks[0].createdBy).toBe('him');
+    expect(tasks[0].status).toBe('open');
+    expect(tasks[0].id).toBeDefined();
+  });
+
+  it('should limit and offset based on parameters', async () => {
+    const { store } = await createStore();
+
+    await store.createTask({
+      spec: {} as TaskSpec,
+      createdBy: 'me',
+    });
+
+    await store.createTask({
+      spec: {} as TaskSpec,
+      createdBy: 'him',
+    });
+
+    const { tasks, totalTasks } = await store.list({
```

Comment:
Same as in StorageTaskBroker it's nice to cover with tests all new introduced options